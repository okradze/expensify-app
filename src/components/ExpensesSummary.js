import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses';
import getExpensesTotal from '../selectors/expenses-total';

export const ExpensesSummary = ({ count, total }) => {
    return (
        <div>
            <p>Viewing {count} {count > 1 ? 'expenses' : 'expense'} totaling {numeral(total / 100).format('$0,0.00')}</p>
        </div>
    );
};

const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters);

    return {
        count: visibleExpenses.length,
        total: getExpensesTotal(visibleExpenses)
    };
};

export default connect(mapStateToProps)(ExpensesSummary);