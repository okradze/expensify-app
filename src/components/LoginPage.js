import React from 'react';
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth';
import google from '../../images/google.svg';

export const LoginPage = ({ startLogin }) => (
    <div className="box-layout">
        <div className="box-layout__box">
            <h1 className="box-layout__title">Expensify App</h1>
            <p>Get your expenses under control.</p>
            <button className="button button--blue" onClick={startLogin}><img className="box-layout__google" src={google}></img>Log in by Google</button>
        </div>
        <p className="box-layout__copyright">Tbilisi's photo was taken by Giorgi Gvilava</p>
    </div>
);

const mapDispatchToProps = (dispatch) => ({
    startLogin: () => dispatch(startLogin())
});
export default connect(undefined, mapDispatchToProps)(LoginPage);