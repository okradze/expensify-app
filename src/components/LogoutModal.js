import React from 'react';
import Modal from 'react-modal';
import { connect } from 'react-redux';
import { startLogout } from '../actions/auth';

export const LogoutModal = ({ isModalOpen, closeModal, startLogout }) => (
    <Modal
        isOpen={isModalOpen}
        onRequestClose={closeModal}
        contentLabel="Log Out"
        appElement={document.getElementById('app')}
        className="modal">


        <h2>Are you sure to log out?</h2>
        <button className="button button--blue modal__button" onClick={closeModal}>Stay</button>
        <button className="button button--red" onClick={startLogout}>Log Out</button>
    </Modal>
);

const mapDispatchToProps = (dispatch) => ({
    startLogout: () => dispatch(startLogout())
})

export default connect(undefined, mapDispatchToProps)(LogoutModal);