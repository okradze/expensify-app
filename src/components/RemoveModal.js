import React from 'react';
import Modal from 'react-modal';

export const RemoveModal = ({ isModalOpen, changeModalState, onRemove }) => (
    <Modal
        isOpen={isModalOpen}
        onRequestClose={changeModalState}
        contentLabel="Remove expense"
        appElement={document.getElementById('app')}
        className="modal" >

        <h2>Are you sure to remove an expense?</h2>
        <button className="button button--blue modal__button" onClick={changeModalState}>Never mind</button>
        <button className="button button--red" onClick={onRemove}>Sure</button>
    </Modal>
);

export default RemoveModal;