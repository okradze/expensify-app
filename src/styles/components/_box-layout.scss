.box-layout {
    display: flex;
    justify-content: center;
    align-items: center;
    background: linear-gradient(rgba(0,0,0,.5), rgba(0,0,0,.5)),url('../images/bg-hero.jpg') no-repeat center center;
    background-size: cover;
    height: 100vh;
    width: 100vw;

    &__box {
        background-color: fade-out(#fff, .2);
        border-radius: 3px;
        text-align: center;
        padding: $l-size $m-size;
    }
    &__title {
        margin: 0 0 $m-size 0;
        line-height: 1;
    }
    &__copyright {
        position: absolute;
        right: 1.5rem;
        bottom: 1.5rem;
        margin: 0;
        color: #b9b9b9;
        font-style: italic;
    }
    &__google {
        width: 3rem;
        height: 3rem;
        margin-right: 1rem;
    }
    button {
        display: flex;
        align-items: center;
        justify-content: center;
        margin: 0 auto;    
    }
}