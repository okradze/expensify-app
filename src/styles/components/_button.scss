.button {
    display: inline-block;
    padding: $s-size;
    font-size: $font-large;
    font-weight: 300;
    border: none;
    border-radius: 1px;
    text-decoration: none;
    line-height: 1;
    background: none;
    color: #fff;
}
.button--blue {
    background-color: $blue;
}

.button--grey {
    background-color: #888;
}