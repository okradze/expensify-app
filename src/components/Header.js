import React from 'react';
import { Link } from 'react-router-dom';
import LogoutModal from './LogoutModal';

export class Header extends React.Component {
    state = {
        isModalOpen: false
    };
    changeModalState = () => {
        this.setState(() => ({ isModalOpen: !this.state.isModalOpen }));
    };
    render() {
        return (
            <header className="header">
                <div className="content-container">
                    <div className="header__content">
                        <Link className="header__title" to="/dashboard">
                            <h1 className="header__heading">Expensify</h1>
                        </Link>
                        <button className="button" onClick={this.changeModalState}>Log Out</button>
                    </div>
                </div>
                <LogoutModal
                    isModalOpen={this.state.isModalOpen} 
                    closeModal={this.changeModalState} />
            </header>
        );
    }
};

export default Header;