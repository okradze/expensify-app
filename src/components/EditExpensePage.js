import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { startEditExpense, startRemoveExpense } from '../actions/expenses';
import RemoveModal from './RemoveModal';

export class EditExpensePage extends React.Component {
    state = {
        isModalOpen: false
    };
    onSubmit = (updates) => {
        this.props.startEditExpense(this.props.expense.id, updates);
        this.props.history.push('/');
    };
    onRemove = () => {
        this.props.startRemoveExpense(this.props.expense.id);
        this.props.history.push('/');
    };
    changeModalState = () => {
        this.setState(() => ({ isModalOpen: !this.state.isModalOpen }));
    };
    render() {
        return (
            <div>
                <div className="page-header">
                    <div className="content-container">
                        <h1 className="page-header__title">Edit Expense</h1>
                    </div>
                </div>
                <div className="content-container">
                    <ExpenseForm
                        expense={this.props.expense}
                        onSubmit={this.onSubmit} />
                    <button className="button button--grey" onClick={this.changeModalState}>Remove</button>
                </div>
                <RemoveModal 
                    isModalOpen={this.state.isModalOpen}
                    closeModal={this.changeModalState}
                    onRemove={this.onRemove} />
            </div>
        );
    };
};

const mapDispatchToProps = (dispatch) => ({
    startEditExpense: (id, update) => dispatch(startEditExpense(id, update)),
    startRemoveExpense: (id) => dispatch(startRemoveExpense(id))
});

const mapStateToProps = (state, props) => ({
    expense: state.expenses.find(expense => expense.id === props.match.params.id)
});

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);