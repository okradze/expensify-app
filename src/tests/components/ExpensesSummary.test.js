import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('should render component correctly with 1 expense', () => {
    const wrapper = shallow(<ExpensesSummary count={1} total={1000} />);
    expect(wrapper).toMatchSnapshot();
});

test('should render component correctly with multiple expense', () => {
    const wrapper = shallow(<ExpensesSummary count={3} total={10000} />);
    expect(wrapper).toMatchSnapshot();
});