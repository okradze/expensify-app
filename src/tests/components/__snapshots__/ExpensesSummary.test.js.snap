// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render component correctly with 1 expense 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        1
      </span>
       
      expense
       totaling 
      <span>
        $10.00
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button button--blue"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;

exports[`should render component correctly with multiple expense 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        3
      </span>
       
      expenses
       totaling 
      <span>
        $100.00
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button button--blue"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;
